### ISSUE TITLE
dask.imperative.do kwargs support

### ISSUE THREAD

**Issue by metaperture** [blaze/dask#721]
Hey Blaze team! I've just started using Blaze/Dask/Odo, and I love your work.

I came across this issue recently--I would expect the following behavior from dask:
```python
>>> from dask import do
>>> @do
>>> def mysum(*args, **kwargs):
>>>     return sum(args) + sum(kwargs.values())
>>> ten = mysum(1,2,3, four=mysum(2,2))
>>> ten.compute() # currently returns a Value object
10
```

Currently, however, it returns a value object, from which it looks like the kwargs in mysum aren't being considered as dask-able. I do get the expected result when I monkey-patch dask.imperative.applyfunc with the following:
```python
def applyfunc(func, args, kwargs, pure=False):
    """Create a Value by applying a function to args.
    Given a function and arguments, return a Value that represents the result
    of that computation."""

    args, dasks = unzip(map(to_task_dasks, args), 2)
    if kwargs:
        # change tokenize implementation so we don't need this?
        assert 'pure' not in kwargs, 'reserved keyword' 
        task_kwvals, kwdasks = unzip(map(to_task_dasks, kwargs.values()), 2)
        kwargs = dict(zip(kwargs.keys(), task_kwvals))
        dasks += kwdasks
    dasks = flat_unique(dasks)
    name = tokenize(func, *args, pure=pure, **kwargs)
    if kwargs:
        # I have no idea why this works, it could probably be simpler
        dasks.append({name: (apply, func, args,
                             (dict, (list, list(map(list, kwargs.items())))))})
    else:
        dasks.append({name: (func,) + args})
    return Value(name, dasks)
```
I tried it with some additional cleverness to keep the functools.partial speedup when possible, but I think it may overcomplicate things:
```python
    if kwargs:
        split_kwargs = groupby(lambda item: isinstance(item[1], Value),
                               kwargs.items())
        func = partial(func, **dict(split_kwargs.get(False, [])))
        task_kwargs = dict(split_kwargs.get(True, []))
        # change tokenize implementation so we don't need this?
        assert 'pure' not in task_kwargs, 'reserved keyword' 
        task_kwvals, kwdasks = unzip(map(to_task_dasks, task_kwargs.values()), 2)
        kwargs = dict(zip(task_kwargs.keys(), task_kwvals))
        dasks += kwdasks
```

...but the weird reservation for the 'pure' keyword still bugs me a bit. Any ideas for what the least invasive way would be to fix that would be? Or is this entire idea somehow antithetical?

Thanks!

PS: Apologies that this isn't in a pull-request

**Comment by metaperture** [comment]
Actually, a better test case:
```python
from dask import do
@do
def mysum(*args, **kwargs):
    return sum(args) + sum(kwargs.values())
ten = mysum(1,2, mysum(1,2), four=mysum(2,2))
ten.compute() # currently returns a Value object
```
Shows the return format needs to be:
```python
    if kwargs:
        # I have no idea why this works, it could probably be simpler
        dasks.append({name: (apply, func, (tuple, list(args)),
                             (dict, (list, list(map(list, kwargs.items())))))})
```
Again, I still don't entirely know why that works, so I might be overcomplicating things. All hail the cargo cult.

**Comment by jcrist** [comment]
Thanks for the bug report and test case. This should be fixed in #722, please try it out and let me know if it solves your problem.

**Comment by metaperture** [comment]
Wow, that was quick! And it works, thanks!!
